compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1202,7 +1206,8 @@ rustc_queries! {
     /// instead, where the instance is an `InstanceDef::Virtual`.
     query fn_abi_of_fn_ptr(
         key: ty::ParamEnvAnd<'tcx, (ty::PolyFnSig<'tcx>, &'tcx ty::List<Ty<'tcx>>)>
-    ) -> Result<&'tcx abi::call::FnAbi<'tcx, Ty<'tcx>>, ty::layout::FnAbiError<'tcx>> {
+    ) -> Result<abi::call::FnAbi<'tcx, Ty<'tcx>>, ty::layout::FnAbiError<'tcx>> {
```

Comment:
hmm... not sure what is better, doing this or making `FnAbiError` also be behind a `&'tcx` or even going as far as replacing the `String`s inside with `&'tcx str`...